prost-types/src/duration.rs

```diff
@@ -69,6 +69,10 @@ impl Duration {
         result.normalize();
         result
     }
+
+    pub fn is_valid(&self) -> bool {
+        self.seconds >= -315_576_000_000 && self.seconds <= 315_576_000_000
```

Comment:
Sorry if I'm a tourist asking silly questions here, but what does it mean to be a duration consisting of a negative number of seconds?

Reply:
There was a conformance test that checked this behavior... It's basically an api contract / invariant, that is mentioned here: https://github.com/protocolbuffers/protobuf/blob/2ad21af8406316d2b170230f93db7d10c36108dd/src/google/protobuf/duration.proto#L103.